### PR TITLE
reorg fixes

### PR DESCRIPTION
**Pull request by arvidn**
when going through a reorg, maintain all chain state until the very end, when the new fork has been fully validated and added.

This addresses the "in-main-chain" issue gene experienced last week.

### CODE REVIEW

File: chia/consensus/blockchain.py [hunk 1]
```diff
@@ -445,6 +441,12 @@ async def _reconsider_peak(
                         hint_coin_state[key] = {}
                     hint_coin_state[key][coin_id] = latest_coin_state[coin_id]
 
+        # we made it to the end successfully
+        # Rollback sub_epoch_summaries
+        self.__height_map.rollback(fork_height)
+        await self.block_store.rollback(fork_height)
+
+        await self.block_store.rollback(fork_height)
```
Comment (mariano54) [inline chia/consensus/blockchain.py:449]: double block store rollback

**Comment by mariano54** [comment]
Looks good, but we should add both tests:
- Ref test (gene issue)
- Fail during reconsider_peak test

**Review by mariano54 — DISMISSED** [review]

**Comment by arvidn** [comment]
these tests still pass on `main`, so I don't quite know exactly what's going on yet

**Review by mariano54 — APPROVED** [review]

File: chia/consensus/blockchain.py [hunk 2]
```diff
@@ -445,6 +441,10 @@ async def _reconsider_peak(
                         hint_coin_state[key] = {}
                     hint_coin_state[key][coin_id] = latest_coin_state[coin_id]
 
+        # we made it to the end successfully
+        # Rollback sub_epoch_summaries
+        self.__height_map.rollback(fork_height)
```
[the remark this answers is not in the document]
Comment (arvidn) [reply chia/consensus/blockchain.py:446]: I agree, but I think it's a bit more to it than that. The existing code updating the height-to-hash map should probably move to after we commit the DB transaction too. I'd prefer to do this in a separate PR

[the remark this answers is not in the document]
Comment (mariano54) [reply chia/consensus/blockchain.py:446]: sounds good

**Pull request merged**